### ISSUE TITLE
DuckDB code works in IDE (DataGrip), but fails in CLI

### ISSUE THREAD

**Issue by kylelundstedt** [duckdb/duckdb#6134]
### What happens?

I want to read (and write) files from a private S3 bucket in batch.  I'm prototyping the SQL code using an IDE (DataGrip), then I intend to parallelize the execution by using the DuckDB CLI in a shell script.

When I run the basic required SQL code in DataGrip, the code works fine.  When I try to run the exact same code in the DataGrip CLI, I get a 403 (Forbidden) connection error.

### To Reproduce

Reproducing the behavior requires a private AWS S3 bucket, and the appropriate credentials to access that bucket.

Running the following code in my IDE (DataGrip) works flawlessly.

```DataGrip
INSTALL httpfs; LOAD httpfs; INSTALL parquet; LOAD parquet;
SET s3_region='us-west-2';
set S3_ACCESS_KEY_ID='XXXXXX';
set S3_SECRET_ACCESS_KEY='YYYYYY';
SELECT * FROM read_csv_auto('s3://BUCKETNAME/FILENAME.csv.gz') limit 100;
```

Running the exact same code in the DuckDB CLI, from the same folder on the same machine, throws a 403 connection error.

```CLI
❯ duckdb
v0.6.1 919cad22e8
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D INSTALL httpfs; LOAD httpfs; INSTALL parquet; LOAD parquet;
D SET s3_region='us-west-2';
D set S3_ACCESS_KEY_ID='XXXXXX';
D set S3_SECRET_ACCESS_KEY='YYYYYY';
D SELECT * FROM read_csv_auto('s3://BUCKETNAME/FILENAME.csv.gz') limit 100;
Error: Invalid Error: Unable to connect to URL "s3://BUCKETNAME/FILENAME.csv.gz": 403 (Forbidden)
```

### OS:

macOS 13.1

### DuckDB Version:

v0.6.1 919cad22e8

### DuckDB Client:

DuckDB CLI

### Full Name:

Kyle Lundstedt

### Affiliation:

IndustryVault

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Tishj** [comment]
If I would have to guess, I'd say your IDE sets some environment variables that you don't have in the environment that you run the CLI in.
```c++
	static constexpr const char *REGION_ENV_VAR = "AWS_DEFAULT_REGION";
	static constexpr const char *ACCESS_KEY_ENV_VAR = "AWS_ACCESS_KEY_ID";
	static constexpr const char *SECRET_KEY_ENV_VAR = "AWS_SECRET_ACCESS_KEY";
	static constexpr const char *SESSION_TOKEN_ENV_VAR = "AWS_SESSION_TOKEN";
```
These are some environment variables the HTTPFS extension looks for

**Comment by kylelundstedt** [comment]
> If I would have to guess, I'd say your IDE sets some environment variables that you don't have in the environment that you run the CLI in.
> 
> ```c++
> 	static constexpr const char *REGION_ENV_VAR = "AWS_DEFAULT_REGION";
> 	static constexpr const char *ACCESS_KEY_ENV_VAR = "AWS_ACCESS_KEY_ID";
> 	static constexpr const char *SECRET_KEY_ENV_VAR = "AWS_SECRET_ACCESS_KEY";
> 	static constexpr const char *SESSION_TOKEN_ENV_VAR = "AWS_SESSION_TOKEN";
> ```
> 
> These are some environment variables the HTTPFS extension looks for

Hi @Tishj, thanks for the quick reply!
 
To test, I first made sure that AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, and AWS_DEFAULT_REGION were all set as environment variables in the same shell where I ran the DuckDB CLI.  I received the same "403 (Forbidden)" error.

Just in case, I also added S3_REGION, S3_ACCESS_KEY_ID, and S3_SECRET_ACCESS_KEY as environment variables, just in case.  No luck there either.

In either case, it's still confusing to me, since the SQL code specifically uses the "SET" command to make those variables available to the DuckDB CLI.

Let men now if you can suggest anything else for me to try.  Thanks!

**Comment by Tishj** [comment]
Could it be possible that they use different versions of duckdb?
Otherwise could it perhaps be related to some proxy settings in DataGrip vs your cli?

**Comment by kylelundstedt** [comment]
DuckDB CLI is v0.6.1 919cad22e8.  DataGrip is using duckdb_jdbc-0.6.1.jar.
So, the version numbers are identical.

The CLI and DataGrip are running on the same machine, and neither has any proxy settings to my knowledge.

Again, I'm baffled why the CLI isn't running the SQL code that specifically uses the "SET" command to make those variables available.

**Comment by Mause** [comment]
Are you able to reproduce this on another bucket, perhaps, so we can investigate there?

Or run Wireshark with ssl off and investigate what's wrong with the requests?

**Comment by kylelundstedt** [comment]
@Mause I get the same issue regardless of the private bucket I choose.  I've downloaded Wireshark, but have no idea how to use to investigate.

**Comment by kylelundstedt** [comment]
I've also downloaded DBeaver, and had no problem accessing the private S3 buckets.
So, I think the issue is specifically with the CLI.

**Comment by kylelundstedt** [comment]
TL;DR:  upgrade to 0.7; set the AWS_DEFAULT_REGION env var; problem solved!!!

I've upgraded to DuckDB CLI v0.7, and it partially fixed the issue.

The good news is, thanks to #5419, the CLI correctly loads AWS credentials from the appropriate environment variables!  Much appreciated, @satotake!

The bad news is the CLI still threw an error, though a different one this time.

In the examples above, I got the `Error: Invalid Error: Unable to connect to URL "s3://BUCKETNAME/FILENAME.csv.gz": 403 (Forbidden)` message.  Now I got `Error: Unable to connect to URL "s3://BUCKETNAME/issues.csv": 400 (Bad Request)` message.

After looking at the `duckdb_settings()` output carefully, it runs out that my `s3_region` setting was not populated.  Once I set it manually, everything worked great!  

It turns out that the `aws configure export-credentials --format env` command that I used to populate my environment variables does NOT include the AWS_DEFAULT_REGION variable, which it appears is required to correctly populate the `s3_region` setting in DuckDB.

So, if I set `s3_region` manually in DuckDB, or add `export AWS_DEFAULT_REGION=us-west-2` to my environment, everything works great!

```
❯ ./duckdb xxx.db
v0.7.0 f7827396d7
Enter ".help" for usage hints.
D INSTALL httpfs; LOAD httpfs;
D SELECT current_setting('s3_session_token');
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                             current_setting('s3_session_token')                                                              │
│                                                                           varchar                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ IQoJ…  │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
D SELECT current_setting('s3_region');
Error: Invalid Input Error: unrecognized configuration parameter "s3_region"
D SET s3_region='us-west-2';
D SELECT current_setting('s3_region');
┌──────────────────────────────┐
│ current_setting('s3_region') │
│           varchar            │
├──────────────────────────────┤
│ us-west-2                    │
└──────────────────────────────┘
D SELECT * FROM read_csv_auto('s3://BUCKETNAME/issues.csv') limit 100;
┌────────────────┬──────────────┬────────────┬─────────────┬────────────────┬─────────────────────┬───────────────────────┬─────────────┐
│    Issue ID    │ Issue Status │ Experience │ Create Date │ Issue Due Date │ Close / Cancel Date │ Issue Line Of Defense │ Risk Rating │
│    varchar     │   varchar    │  varchar   │    date     │      date      │        date         │        varchar        │   varchar   │
├────────────────┼──────────────┼────────────┼─────────────┼────────────────┼─────────────────────┼───────────────────────┼─────────────┤
│ ISSUE-00011111 │ Active       │            │ 2021-02-26  │                │                     │ 1st Line              │ Medium      │
│ ISSUE-00011112 │ Active       │            │ 2020-12-24  │ 2021-03-31     │                     │ 3rd Line              │ High        │
│ ISSUE-00011113 │ Active       │            │ 2020-06-11  │ 2020-12-27     │                     │ 1st Line              │ Low         │
```
